ipc/common/errors.go

```diff
@@ -0,0 +1,63 @@
+package common
+
+import (
+	"fmt"
+)
+
+// CriticalError signals a critical error
+type CriticalError struct {
+	InnerErr error
```

Comment:
I did custom error definitions like this:

```
// ErrBadArwenArguments signals a critical error
var ErrBadArwenArguments = &CriticalError{InnerErr: fmt.Errorf("bad arguments passed to arwen")}
```

I couldn't find a sane way to define new errors using the dummy `%w` `fmt`. I wanted to have this kind of semantics: `ErrBadArwenArguments is a critical` error instead of `ErrBadArwenArguments wraps a critical error`. I wanted to have the pattern:

Lower layer Thrower:

```
throw ErrBadArwenArguments
```

Upper layer Thrower:

```
catch (exception) {
    throw CriticalError(innerException: exception)
}
```